packages/core/src/model.ts

```diff
@@ -116,6 +114,19 @@ export class ChartModel {
 	 */
 	setData(newData) {
 		const sanitizedData = this.sanitize(Tools.clone(newData));
+
+		// Sort data in user defined order
+		const legendOrder = Tools.getProperty(
+			this.getOptions(),
+			"legend",
+			"order"
+		);
+		if (legendOrder.length) {
+			sanitizedData.sort((dataA, dataB) => 
+				legendOrder.indexOf(dataA.group) - legendOrder.indexOf(dataB.group)
+			);
+		}
```

Comment:
this seems to be ordering the entire data instead of just the legend ordering. Hadn't thought about it this way. What will change now? Vs. only changing legend order...?

Reply:
if we only change the legend order, the chart remains old order, it looks weird for me. 
For example, the legend is `Sold, Misc, Qty, Restocking, More`, 
and the bars display as `Qty, More, Sold, Restocking, Misc`. 
I think they should always keep the same ordering.

Reply:
I'd ask the user who created #587. I think they really just wanted to reverse the order of the legend

Reply:
ok, I'll change it.